pkg/metadata/inventories/inventories.go

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package inventories
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/common"
+)
+
+type checkMetadataCacheEntry struct {
+	LastUpdated           int64
+	CheckInstanceMetadata CheckInstanceMetadata
+}
+
+var (
+	// For testing purposes
+	getLoadedConfigs  = common.AC.GetLoadedConfigs
+	getAllInstanceIDs = common.Coll.GetAllInstanceIDs
+	nowNano           = func() int64 { return time.Now().UnixNano() }
+
+	checkMetadataCache = make(map[string]*checkMetadataCacheEntry) // by check ID
```

Comment:
I will need some changes to BasicCache: https://github.com/DataDog/datadog-agent/pull/4228

And even with that, I'm not sure I can use BasicCache's mutex and last updated timestamp in the case of the `checkMetadataCache` (because it's a two level map, with one last updated timestamp per inner level map, but a mutex only for the outer level map).

Reply:
I approved #4228, if that's still not good enough for you, feel free to just continue using what you have here.

Reply:
Let's merge this as it is, and I'll open a PR to switch to BasicCache if I can make it fit nicely.